src/common/QueryHandler.ts

```diff
@@ -14,62 +14,137 @@
  * limitations under the License.
  */
 
-export interface QueryHandler {
+import { WaitForSelectorOptions, DOMWorld } from './DOMWorld.js';
+import { ElementHandle, JSHandle } from './JSHandle.js';
+
+/**
+ * @internal
+ */
+interface QueryHandler {
```

Comment:
what do you think about calling this type `InternalQueryHandler` to really make it clear that it's not user facing?

Reply:
Sure! considering that they are internal, I suppose we can always change our minds about this as well.